storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4176,12 +4176,13 @@ private void maybeAddPlaceholderSpoutAggStats(TopologyPageInfo topoPageInfo, Sto
      *
      * @param topoPageInfo  topology page info holding bolt AggStats
      * @param topology      storm topology used to get bolt names
+     * @param includeSys    whether to show system bolts
      */
-    private void maybeAddPlaceholderBoltAggStats(TopologyPageInfo topoPageInfo, StormTopology topology) {
+    private void maybeAddPlaceholderBoltAggStats(TopologyPageInfo topoPageInfo, StormTopology topology, boolean includeSys) {
         if (topoPageInfo.get_id_to_bolt_agg_stats().isEmpty()) {
             Map<String, Bolt> bolts = topology.get_bolts();
-            for (Entry<String, Bolt> entry : bolts.entrySet()) {
-                if (Utils.isSystemId(entry.getKey())) {
+            for (String boltName : bolts.keySet()) {
+                if (!includeSys && Utils.isSystemId(boltName) || boltName.equals("__system")) {
```

Comment:
`Utils.isSystemId` already covers `boltName.equals("__system")`, so you just need to call `Utils.isSystemId`.

Reply:
When not using placeholder components, there's some filtering on the **__system** bolt that suppresses it's displaying even when the "Show System Stats" option is selected.

The check for __system is trying to keep the placeholder and non-placeholder output in sync.

I can remove the check if it's decided that's a cleaner way to go.

Reply:
Is your intention `(!includeSys && Utils.isSystemId(boltName)) || boltName.equals("__system"))`? Then let's add bracket there to clarify the intention.

I'm not sure where you pointed out as `some filtering`, but now I think it may be natural to filter out system bolts, as it runs per executor and the main purpose for now is shipping metrics V1 which tracking these bolts may not give meaningful value.

Reply:
Added parentheses for clarity